### PR TITLE
Centralize editing within a single main view

### PR DESCRIPTION
**Pull request by tjennison-google**
This provides a more cohesive user experience and makes better use of the available space.

* Moves cohort groups to a single left pane, allowing consistent access while editing items in the central pane.
* Adds a group editing page with more space to edit groups and their criteria. This removes the need for a separate overview page and also the need for an editing drawer for modifiers.
* Deleted concept.test.tsx. It was fragile and constantly breaking on unrelated changes. Now that the business logic has been separated out was providing minimal benefit anyways and should be replaced with UI tests.

### CODE REVIEW

File: ui/src/criteria/attribute.tsx [hunk 1]
```diff
@@ -323,7 +323,7 @@ function AttributeDetails(props: AttributeDetailsProps) {
     return (
       <>
         {props.data.selected.map(({ value, name }) => (
-          <Stack direction="row" alignItems="baseline" key={Date.now()}>
```
Comment (marikomedlock) [inline ui/src/criteria/attribute.tsx:326]: Why the change from Date -> String?

**Comment by tjennison-google** [comment]
Some screenshots of the UI:
image
image

There are still more changes to come, like removing the duplicated elements, but UI is functional in the new format at this point.

File: ui/src/criteria/attribute.tsx (hunk 1 above)
Comment (tjennison-google) [reply ui/src/criteria/attribute.tsx:326]: This was effectively using a random key before, which worked because the list was never visible while being edited. Now the keys actually need to stable of duplicate UI elements can be created.

**Review by marikomedlock — APPROVED** [review]
Thanks for the screenshots, that's really helpful.
